src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,354 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""Implementation of PCR tests."""
+import json
+import logging
+import os
+import signal
+import time
+import subprocess
+import sas_testcase
+from shapely import ops
+from full_activity_dump_helper import getFullActivityDumpSasUut
+from reference_models.ppa import ppa
+from reference_models.geo import drive, utils
+from util import configurable_testcase, loadConfig, \
+     makePalRecordsConsistent, writeConfig, getCertificateFingerprint
+TEST_CONFIG_COMMAND = 'cd reference_models;python test_config.py'
+SAS_TEST_HARNESS_URL = 'https://test.harness.url.not.used/v1.2'
+
+
+def isPpaWithinServiceArea(pal_records, ppa_zone_geometry):
+  """Check if the ppa zone geometry with in service area then return True.
+
+  Checks the ppa zone geometry's boundary and interior intersect only with the
+  interior of the service area (not its boundary or exterior).
+
+  Args:
+    pal_records: A list of pal records to compute service area based on
+      census_tracts.
+    ppa_zone_geometry: A PPA polygon dictionary in GeoJSON format.
+
+  Returns:
+    A value is the boolean with the value as True if the ppa zone geometry's
+      boundary and interior intersect with in the interior of the service
+      area otherwise value as false.
+
+  """
+  # Get the census tract for each pal record and convert it to Shapely
+  # geometry.
+  census_tracts_for_pal = [
+      utils.ToShapely(drive.census_tract_driver.GetCensusTract(
+          pal['license']['licenseAreaIdentifier'])
+                      ['features'][0]['geometry']) for pal in pal_records]
+  pal_service_area = ops.cascaded_union(census_tracts_for_pal)
+
+  # Convert GeoJSON dictionary to Shapely object.
+  ppa_zone_shapely_geometry = utils.ToShapely(ppa_zone_geometry)
+
+  return ppa_zone_shapely_geometry.buffer(-1e-6).within(pal_service_area)
+
+
+def assertRegConditionalsForPpaRefModel(registration_request,
```

Comment:
Would have been best to use the usual convention:
  registration_requests  ( with s for plural as it is a list)
  condition_registration_datas  (same here, although I know datas is not valid english..)

Reply:
We have changed based on your suggestion   here though assertRegistered and assertRegisteredAndGrant methods are using   the same name conventions., I hope we could track with the separate PR for   doc string uniformity and naming convensions for others places in   Spectrum-Access-System wherever applicable.